### ISSUE TITLE
range of jsd loss in SSLD 

### ISSUE THREAD

**Issue by L1aoXingyu** [PaddlePaddle/PaddleClas#538]
Thanks for your excellent work on distillation. I just read the code of `JSDivLoss` and implement it in pytorch as follow.

```python
def jsdiv_loss(y_s, y_t):
	p_s = F.softmax(y_s, dim=1)
	p_t = F.softmax(y_t, dim=1)
	loss = (F.kl_div(p_s, p_t, reduction="sum") * y_s.shape[1] + \
			F.kl_div(p_t, p_s, reduction="sum") * y_s.shape[1]) / 2.
	loss = loss / y_s.shape[0]
	return loss
```

When I start to train the distill model, I found the jsd loss is negative. I just want to know the correction of this implementation and the range of jsd loss.

Thanks for your time.

**Comment by littletomatodonkey** [comment]
Thanks for your attention, Is it negative for paddle or torch?

**Comment by L1aoXingyu** [comment]
> Thanks for your attention, Is it negative for paddle or torch?

It's negative in torch, so I want to know the range of js_div loss in paddle.

**Comment by littletomatodonkey** [comment]
I reviewed the past logs, they are almost in 0~5.

**Comment by L1aoXingyu** [comment]
I implement it according to paddle version as followed, the `jsd_loss` is about 14.

```python
def _kldiv(self, inputs, targets):
	eps = 1.0e-10
	loss = targets * torch.log((targets + eps) / (inputs + eps)) * inputs.shape[1]
	return loss

def jsdiv_loss(self, y_s, y_t):
	p_s = F.softmax(y_s, dim=1)
	p_t = F.softmax(y_t, dim=1)
	loss = (self._kldiv(p_s, p_t) + self._kldiv(p_t, p_s)) / 2.
	avg_loss = torch.mean(loss)
	return avg_loss
```

I have a question about the `jsd_loss` function, it seems not to be Jensen–Shannon divergence. Because js divergence equals to KL(p || (p+q)/2) + KL(q ||, (p+q)/2), but the code shows KL(p || q) + KL(q || p).

**Comment by littletomatodonkey** [comment]
Yes, it's a little bit difference here, we found that the realization in PaddleClas is slightly better, so we simply use the method.

**Comment by L1aoXingyu** [comment]
Thank you so much!
